### PR TITLE
DSERV-293 Complexes API Endpoints

### PR DESCRIPTION
**Pull request by pedrohr**

### CODE REVIEW

File: data/schema-config.yaml [hunk 1]
```diff
@@ -1161,6 +1161,15 @@ complex:
   label_in_input: complex
   db_collection_name: complexes
   db_collection_per_chromosome: false
+  db_indexes:
```
Comment (mingjiecn) [inline data/schema-config.yaml:1164]: should also add persistent index for complex_source and evidence_code

File: src/routers/genericRouters/routerFuzzy.ts [hunk 2]
```diff
@@ -43,18 +43,21 @@ export class RouterFuzzy extends RouterFilterBy implements Router {
         LIMIT ${page * QUERY_LIMIT}, ${QUERY_LIMIT}
         RETURN { ${this.dbReturnStatements} }
     `
-
     const cursor = await db.query(query)
     return await cursor.all()
   }
 
-  async autocompleteSearch (term: string, page: number, apiReturn: boolean = false, customFilter: string = ''): Promise<any[]> {
+  async autocompleteSearch (term: string, page: number, apiReturn: boolean = false, customFilter: string = '', overWriteSearchField: string = ''): Promise<any[]> {
```
Comment (mingjiecn) [inline src/routers/genericRouters/routerFuzzy.ts:50]: Can you write the function documentation for function usage, parameters, returned value. It is hard for me to understand what is the meaning of some of the parameters. Thank you!

**Review by mingjiecn — CHANGES_REQUESTED** [review]
Please check my comment for changes request. Thank you!

File: data/schema-config.yaml (hunk 1 above)
Comment (pedrohr) [reply data/schema-config.yaml:1164]: they are not being used by any of the endpoints as search parameters

Comment (mingjiecn) [reply data/schema-config.yaml:1164]: Oh, I though filtering by complex_source and evidence_code will need those two fields to be indexed?

Comment (pedrohr) [reply data/schema-config.yaml:1164]: I think we only need to return them

File: src/routers/genericRouters/routerFuzzy.ts (hunk 2 above)
Comment (pedrohr) [reply src/routers/genericRouters/routerFuzzy.ts:58]: added :)

**Review by mingjiecn — APPROVED** [review]

**Pull request merged**